pkg/virt-controller/watch/vmi.go

```diff
@@ -435,9 +435,15 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 		} else {
 			vmiCopy.Status.Phase = virtv1.Pending
 			if vmi.Status.TopologyHints == nil {
-				if topologyHints, err := c.topologyHinter.TopologyHintsForVMI(vmi); err != nil {
-					c.recorder.Eventf(vmi, k8sv1.EventTypeWarning, FailedGatherhingClusterTopologyHints, err.Error())
-					return &syncErrorImpl{err, FailedGatherhingClusterTopologyHints}
+				if topologyHints, tscRequirement, err := c.topologyHinter.TopologyHintsForVMI(vmi); err != nil && tscRequirement != topology.NotRequired {
+					switch tscRequirement {
+					case topology.RequiredForMigration:
+						c.topologyHinter.SetNonMigratableCondition(vmiCopy, conditionManager)
```

Comment:
I would not do it here as the convention is to do it in `calculateLiveMigrationCondition`